docs/java-rest/high-level/apis/scroll.asciidoc

```diff
@@ -4,63 +4,73 @@
+--------------------------------------------------
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-scroll-init]
+--------------------------------------------------
+<1> Create the `SearchRequest` and its corresponding `SearchSourceBuilder`.
+Also optionally set the `size` to control how many results to retrieve at
+a time.
+<2> Set the scroll interval
+<3> Read the returned scroll id, which points to the search context that's
+being kept alive and will be needed in the following search scroll call
+<4> Retrieve the first batch of search hits
+
+==== Retrieve all the relevant documents
 
-Here is an example of a scrolled search:
+As a second step, the received scroll identifier must be set to a
+`SearchScrollRequest` along with a new scroll interval and sent through the
+`searchScroll` method. Elasticsearch returns another batch of results with
+a new scroll identifier. This new scroll identifier can then be used in a
+subsequent `SearchScrollRequest` to retrieve the next batch of results,
+and so on. This process should be repeated in a loop until no more results are
+returned, meaning that the scroll has been exhausted and all the matching
+documents have been retrieved.
 
 ["source","java",subs="attributes,callouts,macros"]
 --------------------------------------------------
-include-tagged::{doc-tests}/SearchDocumentationIT.java[search-scroll-example]
+include-tagged::{doc-tests}/SearchDocumentationIT.java[search-scroll2]
 --------------------------------------------------
-<1> Define a scroll parameter as a `TimeValue` corresponding to one minute
-<2> Create a new `SearchRequest`. See <<java-rest-high-search>>
-for more information on how to build `SearchRequest`.
-<3> Set the `scroll` parameter to the `SearchRequest`
-<4> Execute the `SearchRequest`
-<5> Retrieve the first scroll id
-<6> Retrieve the first batch of search hits
-<7> Iterate until there are no more search hits to process
-<8> Create a new `SearchScrollRequest`
-<9> Set the `scroll` parameter again to tell Elasticsearch to keep the search context
-alive for another minute
-<10> Set the scroll id
-<11> Execute the `SearchScrollRequest` using the Search Scroll API
-<12> Retrieve the next scroll id to use in upcoming requests
-<13> Retrieve the next batch of search hits
-<14> Clear the scroll id using the <<java-rest-high-clear-scroll>>.
+<1> Create the `SearchScrollRequest` by setting the required scroll id and
+the scroll interval
+<2> Read the new scroll id, which points to the search context that's
+being kept alive and will be needed in the following search scroll call
+<3> Retrieve another batch of search hits
+<4>
+
+==== Clear the scroll context
+
+Finally, the last scroll identifier can be deleted using the <<java-rest-high-clear-scroll>>
+in order to release the search context. This happens automatically when the
+scroll expires, but it's good practice to do it as soon as the scroll session
+is completed.
```

Comment:
good point. I didn't add that because it felt like repeating the clear scroll docs, so I thought the link is enough, and the clear scroll code is already in the full example at the end. Do you think that's ok?